packages/lab/src/overlay/Overlay.tsx

```diff
@@ -1,100 +1,158 @@
-    const id = useId();
-    const Window = usePortalWindow();
+    const [showComponent, setShowComponent] = useState(false);
+
+    const {
+      arrowProps,
+      open,
+      onOpenChange,
+      context,
+      floating,
+      reference,
+      getTriggerProps,
+      getOverlayProps,
+      getOverlayPosition,
+    } = useOverlay({
+      open: openProp,
+      placement,
+      onOpenChange: onOpenChangeProp,
+    });
+
+    const triggerRef = useForkRef(
+      // @ts-ignore
+      isValidElement(children) ? children.ref : null,
+      reference
+    );
+
+    const floatingRef = useForkRef<HTMLDivElement>(floating, ref);
+
+    useEffect(() => {
+      if (open && !showComponent) {
+        setShowComponent(true);
+      }
+    }, [open, showComponent]);
 
     const handleCloseButton = () => {
-      onOpenChange?.(false);
+      console.log("close ovl");
+      onOpenChange(false);
     };
 
-    if (!open) {
-      return null;
-    }
-
     return (
-      <Portal>
-        <Window
-          className={clsx(withBaseName(), className, {
-            [withBaseName(variant)]: variant === "secondary",
+      <>
+        {isValidElement(children) &&
+          cloneElement(children, {
+            ...mergeProps(getTriggerProps(), children.props),
+            ref: triggerRef,
           })}
-          id={id}
-          ref={ref}
-          {...rest}
-        >
-          <div
-            className={clsx(withBaseName("content"))}
-            data-testid="overlay-content"
-          >
-            {showClose && (
-              <Button
-                onClick={handleCloseButton}
-                className={withBaseName("close")}
-                variant="secondary"
-              >
-                <CloseIcon
-                  accessible-text="close overlay"
-                  className={withBaseName("closeIcon")}
-                />
-              </Button>
+
+        <FloatingPortal>
+          {/* The provider is needed to support the use case where an app has nested modes. The element that is portalled needs to have the same style as the current scope */}
+          <SaltProvider>
+            {open && showComponent && (
+              <FloatingFocusManager context={context} initialFocus={-1}>
+                <div
+                  ref={floatingRef}
+                  className={clsx(withBaseName(), className)}
+                  style={getOverlayPosition()}
+                  {...getOverlayProps()}
+                  {...rest}
+                >
+                  <div className={withBaseName("container")}>
+                    <div className={withBaseName("content")}>{content}</div>
+                    <Button
+                      onClick={handleCloseButton}
+                      variant="secondary"
+                      className={withBaseName("closeButton")}
+                    >
+                      <CloseIcon
+                        accessible-text="close overlay"
```

Comment:
```suggestion
```
This is left over from previous code, but accessibility wise all icons inside buttons should have aria-hidden applied.